### ISSUE TITLE
Too much on screen

### ISSUE THREAD

**Issue by BenMcLean** [umageddon/namDHC#9]
I have to use Windows 10 at a minimum of 150% DPI scaling at 1080p and even then I can barely read anything at normal screen sizes and have to heavily rely on Windows Magnifier. But even with normal vision, this UI looks like it has way too many things on screen at once. The options go down below the bottom of the screen so that some of the options are inaccessible.

**Comment by BenMcLean** [comment]
It cuts off at the bottom, see?
![image](https://user-images.githubusercontent.com/1747250/181866726-3edc9126-fcf2-4822-b355-1b5567457c4f.png)

**Comment by umageddon** [comment]
Are you scaling the screen differently in some way? I cant reproduce the issue here.

**Comment by BenMcLean** [comment]
> Are you scaling the screen differently in some way? I cant reproduce the issue here.

Both Windows 10's "Make text bigger" and "Make everything bigger" were set to 150% and the resolution was 1080p. I'm not aware of there being more to the scaling settings I had than that.

**Comment by TFWol** [comment]
Just chiming in. I can't reproduce the issue either.
I'm able to see all options just fine with default settings.

**Comment by BenMcLean** [comment]
> Just chiming in. I can't reproduce the issue either. I'm able to see all options just fine with default settings.

Did you actually try it with the DPI scaling I mentioned?

**Comment by TFWol** [comment]
Maybe I didn't understand your original wording.

I read it as you were running the script with those settings because doing it normal wouldn't show the full UI.

If you're running Windows with increased text sizes and such, than it wouldn't be the problem of too many options and more of an AutoHotkey thing. I would change the Issue to a request for DPI scaling support.

In the meantime, you can use hacky workaround I was playing around with that should get you by with at least the important stuff.

What you would probably see with it in place:
![image](https://user-images.githubusercontent.com/9045213/208267141-e1c7bcf6-c3b2-4a96-95f0-6d9dc8d68ee4.png)

I just put `gui 1:-DPIScale` above this line:
https://github.com/umageddon/namDHC/blob/179cf4848c77e1a503262137a9089a0e6d13b777/namDHC.ahk#L1463

Would look like this (has some text truncated and ymmv):
![image](https://user-images.githubusercontent.com/9045213/208267305-258ee050-02ab-4c4b-8b0f-d50026960828.png)

# If you don't know how
If you want, I can create a pull request to this repo and you can use the file/file(s) I generate.
I can't promise I'll keep it up-to-date though; just a one-off thing really.

**Comment by BenMcLean** [comment]
> I read it as you were running the script with those settings because doing it normal wouldn't show the full UI.

No, it's that I have to do that stuff to make the text big enough to read.

> If you're running Windows with increased text sizes and such, than it wouldn't be the problem of too many options and more of an AutoHotkey thing.

It looks like it has too many options for any app to ever have crammed into one screen.

> I would change the Issue to a request for DPI scaling support.

OK.

> In the meantime, you can use hacky workaround I was playing around with that should get you by with at least the important stuff.

I'll try that next time, although I'm not a super heavy user of the app so I don't know when next time will be.

Windows apps need to be DPI-aware, as discussed in this article from Microsoft https://msdn.microsoft.com/en-us/library/windows/desktop/dn469266(v=vs.85).aspx and explained in this tutorial. https://msdn.microsoft.com/en-us/library/windows/desktop/dd464659(v=vs.85).aspx

Should I be suggesting that to AutoHotkey?

**Comment by TFWol** [comment]
> Should I be suggesting that to AutoHotkey?

Not really; it's a tool that lets us run the scripts people create. It depends on each person how much work they want to put into a project.

From the stuff I've dabbled with, making a functional GUI can be a pain in the butt in general.


> Windows apps need to be DPI-aware, as discussed ...

The way I view Github projects like these are as someone like you or me who made something cool (usually something they started for themselves), but who's nice enough to share the tool. 

It's why I had mentioned framing stuff as a request rather than a critique since, in the end, they can do what they want really.
I know one dev (in a certain project I won't name) who refuses to fix a certain feature in his program because someone made him mad 😆 


I went ahead and compiled the code [here ](https://github.com/TFWol/namDHC/blob/main/namDHC.exe) with the small change. Pretty much just toss that in the same directory as the chdman.exe and you'll be good to go.
